src/Umbraco.Web/Editors/MacroController.cs

```diff
@@ -86,11 +86,13 @@ public HttpResponseMessage GetMacroResultAsHtmlForEditor(string macroAlias, int
             UmbracoContext.HttpContext.Items["pageElements"] = legacyPage.Elements;
             UmbracoContext.HttpContext.Items[global::Umbraco.Core.Constants.Conventions.Url.AltTemplate] = null;
 
+            var renderer = new UmbracoComponentRenderer(UmbracoContext);

```

Comment:
If this controller was in a user project than I would say yes! but we can't get ahead of ourselves here, internally Umbraco doesn't use IoC so it can't be 'injected' unless we create a specific controller factory which isn't in the scope of this PR. In Umbraco v8 we will have IoC built in. For now, this PR is about breaking apart UmbracoHelper and allowing better unit-testability for end-users. I'll be pulling this PR in regardless but just wanted to allow feedback on it from the community.